test/vvenc_unit_test/vvenc_unit_test.cpp

```diff
@@ -454,6 +462,80 @@ static bool test_InterPred()
 }
 #endif
 
+#if ENABLE_SIMD_OPT_DIST
+static bool check_lumaWeightedSSE( RdCost* ref, RdCost* opt, unsigned num_cases, int width, int height )
+{
+  printf( "Testing RdCost::lumaWeightedSSE %dx%d\n", width, height );
+
+  std::ostringstream sstm;
+  sstm << "lumaWeightedSSE" << " w=" << width << " h=" << height;
+
+  bool passed = true;
+  for( unsigned i = 0; i < num_cases; i++ )
+  {
+    DimensionGenerator rng;
+    int stride = rng.get( 1, 1024 );
+    int size = std::max( stride, width );
+    std::vector<Pel> orgBuf( size * height );
+    std::vector<Pel> curBuf( size * height );
+    std::vector<Pel> orgLumaBuf( size * height * 4 );
+    std::vector<uint32_t> lumaWeights( 1024 );
+
+    DistParam dtParam;
+    dtParam.org.buf = orgBuf.data();
+    dtParam.cur.buf = curBuf.data();
+    dtParam.org.width = width;
+    dtParam.org.height = height;
+    dtParam.cur.stride = stride;
+    dtParam.org.stride = stride;
+    CPelBuf pelBuf;
+    pelBuf.buf = orgLumaBuf.data();
+    pelBuf.stride = stride;
+    dtParam.orgLuma = &pelBuf;
+    dtParam.bitDepth = 10;
+    dtParam.compID = i % 2 ? COMP_Y : COMP_Cb;
+
+    InputGenerator<int16_t> g14( 14 );
+    InputGenerator<uint16_t> g10( 10 ); // Index range : 0 - 1023.
+    InputGenerator<uint32_t> g18( 17 );
+    std::generate( orgBuf.begin(), orgBuf.end(), g14 );
+    std::generate( curBuf.begin(), curBuf.end(), g14 );
+    std::generate( orgLumaBuf.begin(), orgLumaBuf.end(), g10 );
+    std::generate( lumaWeights.begin(), lumaWeights.end(), g18 );
+    for( unsigned csx = 0; csx < 2; csx++ )
+    {
+      Distortion sum_ref = ref->m_wtdPredPtr[csx]( dtParam, VVENC_CHROMA_420, lumaWeights.data() );
+      Distortion sum_opt = opt->m_wtdPredPtr[csx]( dtParam, VVENC_CHROMA_420, lumaWeights.data() );
+      passed = compare_value( sstm.str(), sum_ref, sum_opt ) && passed;
+    }
+  }
+
+  return passed;
+}
+
+static bool test_RdCost()
+{
+  RdCost ref;
+  RdCost opt;
+  ref.create( /*enableOpt=*/false );
+  opt.create( /*enableOpt=*/true );
+
+  unsigned num_cases = NUM_CASES;
+  bool passed = true;
+  std::array<int, 8> widths = { 1, 2, 4, 8, 16, 32, 64, 128 };
+  std::array<int, 7> heights = { 2, 4, 8, 16, 32, 64, 128 };
```

Comment:
Not sure blocks of width 1 are possible, and luma blocks of width or height 2 either. Also, no 4x4, 4x8 or 8x4 luma BI-predicted blocks are possible.

Reply:
Forget this comment - I see this function is also used during search and also for funny sizes like 1xN.